codeflash/api/aiservice.py

```diff
@@ -141,20 +146,19 @@ def optimize_python_code(  # noqa: D417
         payload = {
             "source_code": source_code,
             "dependency_code": dependency_code,
-            "num_variants": num_candidates,
             "trace_id": trace_id,
             "python_version": platform.python_version(),
             "experiment_metadata": experiment_metadata,
             "codeflash_version": codeflash_version,
             "current_username": get_last_commit_author_if_pr_exists(None),
             "repo_owner": git_repo_owner,
             "repo_name": git_repo_name,
-            "n_candidates": N_CANDIDATES_EFFECTIVE,
             "is_async": is_async,
+            "model": model,
```

Comment:
i would recommend we do model selection from our backend. that way we can keep switching models etc more easily.